packages/flutter/lib/src/material/search.dart

```diff
@@ -501,51 +514,47 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
       case null:
         break;
     }
-    final String routeName;
+    String? routeName;
     switch (theme.platform) {
-      case TargetPlatform.iOS:
-      case TargetPlatform.macOS:
-        routeName = '';
-        break;
       case TargetPlatform.android:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
         routeName = searchFieldLabel;
+        break;
```

Comment:
All of these changes to the switch statement are unrelated to the decoration changes, so we shouldn't modify it here.